data-sources/sourcecred/src/generate-datasets.ts

```diff
@@ -0,0 +1,205 @@
+
+    let participant: CredGrainViewParticipantPlusWallet[] | undefined =
+      peopleWhoDidtuffMap.get(signerAddress)
+
+    console.info(
+      `${passportId} linked to wallet? ${participant ? ' - YES' : ' - NO'}`
+    )
+
+    intervalsWeCareAbout.forEach((interval) => {
+      const csvRow = {
+        week_end: interval[0].endDate,
+        sourcecred_score: participant
+          ? sumCredForIdentities(interval, participant)
+          : 0,
+      }
+      csvRows.push(csvRow)
+    })
+
+    writer.writeRecords(csvRows)
+  }
+}
+
+function sumCredForIdentities(
+  interval: [Interval, number],
+  participant: CredGrainViewParticipantPlusWallet[]
+): number {
+  return participant.reduce(
+    (acc: number, el: CredGrainViewParticipantPlusWallet) => {
+      return el.credPerInterval[interval[1]] > 0.01
+        ? acc + el.credPerInterval[interval[1]]
+        : acc
+    },
+    0
+  )
+}
+
+function getParticipantsWhoParticipated(
+  ledger: any,
+  credGrainView: any
+): Map<string, CredGrainViewParticipantPlusWallet[]> {
+  console.info(`getParticipantsWhoParticipated`)
+  const peopleWhoDidStuff: CredGrainViewParticipantPlusWallet[] = []
+  credGrainView
+    .participants()
+    .forEach((participant: CredGrainViewParticipantPlusWallet) => {
+      const anyCred = participant.credPerInterval.some((cred) => cred > 0.1)
+      if (anyCred) {
+        console.info(
+          `Participant ${participant.identity.name} participated, trying to fetch their wallet address`
+        )
+        const account = ledger.accountByName(participant.identity.name)
+        if (account.payoutAddresses.size > 0) {
+          participant.walletAddress = Array.from(
+            account.payoutAddresses.values()
+          )
+        }
+        peopleWhoDidStuff.push(participant)
+      }
+    })
+
+  console.info(
+    `${peopleWhoDidStuff.length} active participants ever (Cred greater than 0.1 in any 1 week )`
+  )
+
+  console.info(`Creating a Map keyed on wallet addresses`)
+
+  const peopleWhoDidtuffMap = peopleWhoDidStuff.reduce(
+    (
+      acc: Map<string, CredGrainViewParticipantPlusWallet[]>,
+      el: CredGrainViewParticipantPlusWallet
+    ) => {
+      el.walletAddress?.forEach((address: string) => {
+        acc.get(address) ? acc.get(address)?.push(el) : acc.set(address, [el])
+      })
+      return acc
+    },
+    new Map<string, CredGrainViewParticipantPlusWallet[]>()
+  )
+
+  console.info(
+    `Returning ${peopleWhoDidtuffMap.size} participants who have wallet addresses in SourceCred`
+  )
+
+  return peopleWhoDidtuffMap
+}
+
+function buildIntervals(credGrainView: any): Array<[Interval, number]> {
+  //intervals in SC are 1 week and run from sunday midnight UTC to sunday midnight UTC
+  //the intervals list in the credGrainView facilitates indexing into the cred array on
+  //each participant
+  //we are interested in data since 29/05/2022
+  const startDate = new Date('2022-05-29').setUTCHours(0, 0, 0, 0)

```

Comment:
```suggestion
  const startDate: Date = new Date('2022-05-29').setUTCHours(0, 0, 0, 0)

```

Not sure if this suggested change will make the log output more human-readable: "Building interval map starting at week ending 1653782400000"

Reply:
It won't but I can make it more human, I knew I had done that, lazy ;)